File: tests/snapshots/string_funcs.rs
// Built-in string functions

fn main() {
    let s: String = "Hello, World".to_string();
    println!("length:    {}", s.len());
    println!("upper:     {}", s.to_uppercase());
    println!("lower:     {}", s.to_lowercase());
    println!("left 5:    {}", s.chars().take(5).collect::<String>());
    println!("right 5:   {}", s.chars().skip(s.chars().count().saturating_sub(5)).collect::<String>());
    println!("mid 2,3:   {}", s.chars().skip(1).take(3).collect::<String>());
    println!("trimmed:   {}", "   padded   ".trim());
    println!("replaced:  {}", s.replace("World", "Rust"));
    println!("str of 42: {}", 42.to_string());
}

File: tests/snapshots/string_funcs.py
# Built-in string functions

def _vb(x):
    if isinstance(x, bool):
        return "true" if x else "false"
    if isinstance(x, float) and x.is_integer():
        return str(int(x))
    return str(x)

def main():
    s: str = 'Hello, World'
    print(f"length:    {_vb(len(s))}")
    print(f"upper:     {_vb((s).upper())}")
    print(f"lower:     {_vb((s).lower())}")
    print(f"left 5:    {_vb((s)[:max(int(5), 0)])}")
    print(f"right 5:   {_vb(((s)[-int(5):] if int(5) else ''))}")
    print(f"mid 2,3:   {_vb((s)[max(int(2) - 1, 0):max(int(2) - 1, 0) + int(3)])}")
    print(f"trimmed:   {_vb(('   padded   ').strip())}")
    print(f"replaced:  {_vb((s).replace('World', 'Rust'))}")
    print(f"str of 42: {_vb(str(42))}")


if __name__ == "__main__":
    main()
